### PR TITLE
[Hotfix][Core] Fix the NullPointException for the json config of the job without pluginname

### PR DESCRIPTION
**Pull request by dailai**
### Purpose of this pull request
When we use the json config of the job define that without pluginname, this case will throw NullPointerException.
截屏2024-05-09 08 17 40

Like this:
`{
  "env": {
    "parallelism": 4,
    "job.mode": "BATCH"
  },
  "source": [
    {
      "result_table_name": "fake",
      "row.num": 100,
      "schema": {
        "fields": {
          "name": "string",
          "age": "int",
          "card": "int"
        }
      }
    }
  ],
  "transform": [
  ],
  "sink": [
    {
      "source_table_name": "fake1"
    }
  ]
}
`

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by Hisoka-X — COMMENTED** [review]
I think we should check flink and spark job too.

**Comment by dailai** [comment]
> I think we should check flink and spark job too.

Yes, 

> I think we should check flink and spark job too.

That's right. I haven't committed my code yet

File: seatunnel-core/seatunnel-flink-starter/seatunnel-flink-15-starter/src/test/java/org/apache/seatunnel/core/starter/flink/multitable/MultiTableSinkTest.java [hunk 1]
```diff
@@ -34,6 +35,7 @@
 import java.util.Collections;
 import java.util.List;
 
+@Order(1)
```
Comment (Hisoka-X) [inline seatunnel-core/seatunnel-flink-starter/seatunnel-flink-15-starter/src/test/java/org/apache/seatunnel/core/starter/flink/multitable/MultiTableSinkTest.java:38]: Any reason for changed order?

Comment (dailai) [reply seatunnel-core/seatunnel-flink-starter/seatunnel-flink-15-starter/src/test/java/org/apache/seatunnel/core/starter/flink/multitable/MultiTableSinkTest.java:38]: https://github.com/dailai/seatunnel/actions/runs/9015097694/job/24775117908

Because both use the InMemory sink. This may lead your test be failed.
截屏2024-05-11 13 38 55

**Review by EricJoy2048 — APPROVED** [review]

File: seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties [hunk 2]
```diff
@@ -0,0 +1,20 @@
+################################################################################
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: Is this file necessary?

Comment (dailai) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: Yes, otherwise the Order on the class does not take effect

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: I feel a little confused. The scope of this file and the test case for adding `order` are not in the same package. Why do they affect each other?

Comment (dailai) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: This is the feature of Junit5. The test case depend on the module of e2e-common. The file takes effect as long as you configure junit-platform.properties in the resource directory under test.

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: Oh. I got it.

Comment (dailai) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/resources/junit-platform.properties:1]: Reference:
https://junit.org/junit5/docs/current/user-guide/#writing-tests-test-execution-order-classes.

image

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @dailai !
